comp/core/tagger/impl-remote/remote.go

```diff
@@ -269,65 +272,78 @@ func (t *remoteTagger) GenerateContainerIDFromOriginInfo(originInfo origindetect
 		}
 	}()
 
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 500 * time.Millisecond
-	expBackoff.MaxInterval = 1 * time.Second
-	expBackoff.MaxElapsedTime = 15 * time.Second
-
-	var containerID string
+	// Generate cache key
+	initPrefix := ""
+	if originInfo.ExternalData.Init {
+		initPrefix = "i/"
+	}
+	key := cache.BuildAgentKey(
+		"remoteTagger",
+		"container_id",
+		initPrefix+originInfo.ExternalData.PodUID+"/"+originInfo.ExternalData.ContainerName,
+	)
 
-	err := backoff.Retry(func() error {
-		select {
-		case <-t.ctx.Done():
-			return &backoff.PermanentError{Err: errTaggerFailedGenerateContainerIDFromOriginInfo}
-		default:
-		}
+	cachedContainerID, err := cache.GetWithExpiration(key, func() (containerID string, err error) {
+		expBackoff := backoff.NewExponentialBackOff()
+		expBackoff.InitialInterval = 500 * time.Millisecond
```

Comment:
maybe we can start by 200 instead of 500

Reply:
Done 👍